iroh-bytes/src/store/flat.rs

```diff
@@ -591,39 +653,31 @@ impl Map for Store {
     type Entry = Entry;
     type Outboard = PreOrderOutboard<MemOrFile>;
     type DataReader = MemOrFile;
-    fn get(&self, hash: &Hash) -> Option<Self::Entry> {
-        let state = self.0.state.read().unwrap();
-        if let Some(entry) = state.complete.get(hash) {
-            state.get_entry(hash, entry, &self.0.options)
-        } else if let Some(entry) = state.partial.get(hash) {
-            let data_path = self.0.options.partial_data_path(*hash, &entry.uuid);
-            let outboard_path = self.0.options.partial_outboard_path(*hash, &entry.uuid);
-            tracing::trace!(
-                "got partial: {} {} {}",
-                hash,
-                entry.size,
-                hex::encode(entry.uuid)
-            );
-            Some(Entry {
-                hash: blake3::Hash::from(*hash),
-                is_complete: false,
-                entry: EntryData {
-                    data: Either::Right((data_path, entry.size)),
-                    outboard: Either::Right(outboard_path),
-                },
-            })
-        } else {
-            tracing::trace!("got none {}", hash);
-            None
-        }
+
+    fn get(&self, hash: &Hash) -> io::Result<Option<Self::Entry>> {
+        self.try_get(hash).map_err(to_io_err)
     }
 }
 
 impl ReadableStore for Store {
-    fn blobs(&self) -> Box<dyn Iterator<Item = Hash> + Send + Sync + 'static> {
-        let inner = self.0.state.read().unwrap();
-        let items = inner.complete.keys().copied().collect::<Vec<_>>();
-        Box::new(items.into_iter())
+    fn blobs(&self) -> io::Result<DbIter<Hash>> {
+        // TODO: should return Result
```

Comment:
this is fixed, so no need to return empty iters on error anymore, should return the errors